Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -80,6 +80,45 @@ def cmd_fail(*args)
     output
   end
 
+  def include_formula_file
+    @formula_file = CoreTap.new.formula_dir/"testball.rb"
+    @content = <<-EOS.undent
+      class Testball < Formula
+        desc "Some test"
+        homepage "https://example.com/testball"
+        url "file://#{File.expand_path("..", __FILE__)}/tarballs/testball-0.1.tbz"
+        sha256 "#{TESTBALL_SHA256}"
+
+        option "with-foo", "Build with foo"
+
+        def install
+          (prefix/"foo"/"test").write("test") if build.with? "foo"
+          prefix.install Dir["*"]
+        end
+
+        # something here
+      end
+    EOS
+    @formula_file.write @content
```

Comment:
Yes, preferably just `content` to avoid creating too many harder-to-keep-track-of instance variables. If a test really needs to retrieve the formula text, it can easily do `formula_path.read` (assuming `formula_path` is the local variable name of the `Pathname` instance representing the created formula).